agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java

```diff
@@ -314,24 +314,27 @@ private static Object callDistributedFileSystem(@SuperCall Callable<?> zuper, @T
 
     private static Object executeMethod(@SuperCall Callable<?> zuper, String uri, String src, String dst, String fsAction, String username) throws Exception {
         long startTime = System.nanoTime();
+        DataAccessEventProtos.FsEvent.Status status = DataAccessEventProtos.FsEvent.Status.SUCCESS;
         try {
             return zuper.call();
         } catch (Exception e) {
+            status = DataAccessEventProtos.FsEvent.Status.FAILURE;
             throw e;
         } finally {
             long elapsedTime = (System.nanoTime() - startTime) / NANOSECONDS_PER_MILLISECOND;
-            sendFsEvent(uri, src, dst, fsAction, username, elapsedTime);
+            sendFsEvent(uri, src, dst, fsAction, username, elapsedTime, status);
```

Comment:
do we want to send failed events?

Reply:
yes because Garmadon's output is not only datalineage. It's also used to detect response time of the namenode from client perspective, iops per app/user, etc...